components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -212,31 +214,43 @@ private IWAAuthenticatedUserBean userInformationInListedUserStores(String authen
                                                                        String userStoreDomains)
             throws AuthenticationFailedException {
 
-        String userStoreDomainForUser = null;
         IWAAuthenticatedUserBean authenticatedUserBean = new IWAAuthenticatedUserBean();
-        try {
-            for (String userStoreDomain : userStoreDomains.split(",")) {
-                resolveUserFromUserStore(
-                        authenticatedUserBean, authenticatedUserName, tenantDomain, userStoreDomain.trim());
-                if (authenticatedUserBean.isUserExists()) {
-                    userStoreDomainForUser = userStoreDomain.trim();
-                    break;
+        authenticatedUserBean.setTenantDomain(tenantDomain);
+        authenticatedUserBean.setUser(authenticatedUserName);
+        MultiAttributeLoginService multiAttributeLoginService =
+                IWAServiceDataHolder.getInstance().getMultiAttributeLoginService();
+
+        if (multiAttributeLoginService.isEnabled(tenantDomain)) {
+            List<String> userStoreDomainsList = Arrays.stream(userStoreDomains.toUpperCase().split(","))
```

Comment:
Is userStoreDomains null safe ?

Reply:
yes, https://github.com/wso2-extensions/identity-local-auth-iwa-kerberos/blob/6f93b37a06ebaf7283992851d1692a50f00853bd/components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java#L115